src/js/client/ajaxCapable.js

```diff
@@ -0,0 +1,142 @@
+ `jQuery.ajax()`.  Once the request is sent, the following workflow applies:
+
+   1.  If the AJAX request returns an error or the result contains a "falsy" `ok` variable, the error is transformed
+       using the rules found in `options.rules.error`, and the results are applied to the component's model using the
+       change applier.
+   2.  If the AJAX request is successful, the results are transformed using the rules found in `options.rules.success`,
+       and applied to the model using the change applier.
+
+ This component does not handle any rendering, you are expected to do that yourself, or use a grade that handles that.
+ With that in mind, there are grades for two common use cases:
+
+   1.  Retrieve required data on startup and then render the page.
+   2.  Submit the model to a REST endpoint and refresh the model with the results.
+
+For the first use case, you can start with the 'retrieveAndRender` component included with this package.
+
+For the second use case, you can start with the `templateFormControl` component.
+
+If you need to do both (or each multiple times), you should create a parent component that uses as many individual
+`templateRequestAndRender` and `templateFormControl` components as needed.
+
+ */
+/* global fluid, jQuery */
+(function ($) {
+    "use strict";
+    var gpii = fluid.registerNamespace("gpii");
+    fluid.registerNamespace("gpii.templates.ajaxCapable");
+
+
+    // TODO: Replace this with JSON Schema validation: https://issues.gpii.net/browse/GPII-1176
+    gpii.templates.ajaxCapable.checkRequirements = function (that) {
+        var errors = [];
+
+        if (!that.options.rules || !that.options.rules.error || !that.options.rules.success || !that.options.rules.request) {
+            errors.push("You have not configured the required rules regarding how to handle AJAX responses.");
+        }
+
+        if (errors.length > 0) {
+            fluid.fail(errors);
+        }
+    };
+
+    gpii.templates.ajaxCapable.makeRequest = function (that) {
+        var options = fluid.copy(that.options.ajaxOptions);
+
+        var transformedModel = fluid.model.transformWithRules(that.model, that.options.rules.request);
+
+        // We have added a `json` option not supported by `jQuery.ajax()` itself, which makes it easier to pass JSON data.
+        if (options.json) {
+            options.contentType = "application/json";
+        }
+
+        options.data = options.json ? JSON.stringify(transformedModel) : transformedModel;
+        $.ajax(options);
+    };
+
+    gpii.templates.ajaxCapable.handleSuccess = function (that, data) {
+        // We assume that we are working with a success unless we have explicit data that suggests otherwise.
+        var okData = fluid.model.transformWithRules(data, that.options.rules.ok);
+        if (okData.ok === undefined || okData.ok === null || okData.ok) {
```

Comment:
Hi, @amb26, this is explicitly needed for interfaces that serve up an error message with a 200 status code.  None of my interfaces do this, but some we use (notably EASTIN) do this.  Rather than enshrine handling for this kind of crappiness, I can encapsulate this check as an extended grade in that code and trust that we are handling errors with `handleError` and success with `handleSuccess`.